pdata/pprofile/attributes.go

```diff
@@ -26,3 +30,41 @@ func FromAttributeIndices(table AttributeTableSlice, record attributable) pcommo
 
 	return m
 }
+
+// AddAttribute updates an AttributeTable and a record's AttributeIndices to
+// add a new attribute.
+// The record can by any struct that implements an `AttributeIndices` method.
+func AddAttribute(table AttributeTableSlice, record attributable, key string, value any) error {
+	for i := range table.Len() {
+		a := table.At(i)
+
+		if a.Key() == key && value == a.Value().AsRaw() {
```

Comment:
nit: I know that `AsRaw()` could make a copy of the data. This is a reminder that it would be nice if you could write `a.Value().Equal(value)` and avoid the allocation.

Reply:
@dmathieu Happy to merge after you reply to this :)

Reply:
I have opened https://github.com/open-telemetry/opentelemetry-collector/issues/12561 to address this in another PR, since it requires changed in `pcommon`.